tools/ctl/ctl.c

```diff
@@ -15,6 +16,29 @@
 #include "ipc/stream.h"
 #include "ipc/control.h"
 
+#define BUFFER_TAG_OFFSET	0
+#define BUFFER_SIZE_OFFSET	1
+#define BUFFER_ABI_OFFSET	2
+
+struct ctl_data {
+	/* the input file name */
+	char *input_file;
+	/* point to the cached data for input/output */
+	unsigned int *buffer;
+	/* flag for input/output format, binary or CSV */
+	bool binary;
+	/* flag for input/output format, with or without abi header */
```

Comment:
best to also have file descriptor for input file and output file here. This could also be stdin and stdout.

Reply:
addressed.